src/main/NWNX/API/Events/BarterEvents.cs

```diff
@@ -0,0 +1,60 @@
+using NWN.API;
+using NWN.Core;
+using NWN.Core.NWNX;
+
+namespace NWNX.API.Events {
+  public class BarterEvents {
+    [NWNXEvent("NWNX_ON_BARTER_START_BEFORE")]
+    public class OnBarterStartBefore : EventSkippable<OnBarterStartBefore>
+    {
+      public NwPlayer Player { get; private set; }
+      public NwPlayer Target { get; private set; }
+
+      protected override void PrepareEvent(NwObject objSelf) {
+        Player = (NwPlayer) objSelf;
+        Target = NWScript.StringToObject(EventsPlugin.GetEventData("BARTER_TARGET")).ToNwObject<NwPlayer>();
+      }
+    }
+    
+    [NWNXEvent("NWNX_ON_BARTER_START_AFTER")]
+    public class OnBarterStartAfter : EventSkippable<OnBarterStartAfter>
+    {
+      public NwPlayer Player { get; private set; }
+      public NwPlayer Target { get; private set; }
+
+      protected override void PrepareEvent(NwObject objSelf) {
+        Player = (NwPlayer) objSelf;
+        Target = NWScript.StringToObject(EventsPlugin.GetEventData("BARTER_TARGET")).ToNwObject<NwPlayer>();
+      }
+    }
+    
+    [NWNXEvent("NWNX_ON_BARTER_END_BEFORE")]
+    public class OnBarterEndBefore : EventSkippable<OnBarterEndBefore>
+    {
+      public NwPlayer Player { get; private set; }
+      public NwPlayer Target { get; private set; }
+      public bool Completed { get; private set; }
+      public int InitiatorItemCount { get; private set; }
+      public int TargetItemCount { get; private set; }
+      public NwObject[] PlayerItems { get; private set; }
+      public NwObject[] TargetItems { get; private set; }
```

Comment:
These are not set.

Reply:
Oops didn't mean to push that file yet. Was trying to determine the best way to create the collection.

Reply:
I removed the event in question for the time being. Will add back later.